priam/src/main/java/com/netflix/priam/notification/BackupNotificationMgr.java

```diff
@@ -102,6 +107,42 @@ private Map<String, MessageAttributeValue> getMessageAttributes(
                 new MessageAttributeValue()
                         .withDataType("String")
                         .withStringValue(backupFileType.name()));
+
+        String additionalAttrs =
+                this.backupRestoreConfig.getBackupNotificationAdditionalMessageAttrs();
+        if (!StringUtils.isBlank(additionalAttrs)) {
+            try {
+                List<String> attrs =
+                        Splitter.on(",")
+                                .omitEmptyStrings()
+                                .trimResults()
+                                .splitToList(additionalAttrs);
+                for (String attr : attrs) {
+                    Object value = additionalAttributes.get(attr);
+                    if (value != null) {
+                        // Does not handle all data types
+                        if ("Instant".equals(value.getClass().getSimpleName())) {
+                            byte[] byteValue = SerializationUtils.serialize((Instant) value);
+                            messageAttributes.putIfAbsent(
+                                    attr,
+                                    new MessageAttributeValue()
+                                            .withDataType("Binary")
+                                            .withBinaryValue(ByteBuffer.wrap(byteValue)));
+                        } else {
+                            messageAttributes.putIfAbsent(
+                                    attr,
+                                    new MessageAttributeValue()
+                                            .withDataType("String")
+                                            .withStringValue(value.toString()));
+                        }
+                    }
+                }
+            } catch (Exception e) {
```

Comment:
I'm not sure what you're expecting to catch here. Please remove the try-catch.

Reply:
I was just trying to be safe. Didn't want any bug or issue in parsing config to cause the notifications to be not sent.

Reply:
Will remove it.